azurerm/internal/services/mssql/helper/sql_extended_auditing.go

```diff
@@ -9,14 +9,17 @@ import (
 
 func ExtendedAuditingSchema() *schema.Schema {
 	return &schema.Schema{
-		Type:     schema.TypeList,
-		Optional: true,
-		MaxItems: 1,
+		Type:       schema.TypeList,
+		Optional:   true,
+		Computed:   true,
+		Deprecated: "all `extended_auditing_policy` properties have been moved to `azurerm_mssql_server_extended_auditing_policy` and `azurerm_mssql_database_extended_auditing_policy`. This block will be removed in version 3.0 of the provider.",
+		ConfigMode: schema.SchemaConfigModeAttr,
+		MaxItems:   1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
 				"storage_account_access_key": {
 					Type:         schema.TypeString,
-					Required:     true,
+					Optional:     true,
```

Comment:
Because `storage_account_access_key` will become optional in `azurerm_mssql_database_extended_auditing_policy`. Once linked to a storage account behind firewall, we'll need assign a role to the sql server, and the `storage_account_access_key` must not be assigned. According to doc https://docs.microsoft.com/en-us/azure/azure-sql/database/audit-write-storage-account-behind-vnet-firewall

Reply:
But this resource should be used after the role is assigned? i'm not sure i follow why this is now optional for the resource..